[EDU-3124] feat: add configure main settings (#431)
aziontech/docs: src/content/docs/en/pages/secure-journey/edit-edge-firewall/edit-main-settings.mdx

---

> title: Configure main settings

Author: ```suggestion
title: How to configure main settings
```

description: Customize your edge firewall domains and modules.
meta_tags: secure, edge firewal, configuration, settings, domains
namespace: docs_guides_secure_configure_main_settings
permalink: /documentation/products/guides/secure/configure-main-settings/
menu_namespace: secureMenu
---

import Button from '~/components/Button.astro'
import Badge from '~/components/Badge.astro'

> Once you create an edge firewall with Azion, you can modify a few configurations. In this guide, you'll learn to modify the domains associated to your firewall and to enable the available modules.

Reviewer: ```suggestion
Once you [create an edge firewall]() with Azion, you can modify a few configurations. In this guide, you'll learn to modify the domains associated with your firewall and to enable the available modules.
```

Reviewer: suggestion: add link to guide.

> 
> ## Via RTM

Reviewer: ```suggestion

---

## Via RTM
```

In this section, you'll modify the main settings of your edge firewalls: name, domains, and modules via [Real-Time Manager (RTM)](https://manager.azion.com).

[Access RTM](/en/documentation/products/guides/how-to-access-rtm/) > select **Edge Firewall**.

1. From the list, select the edge firewall you want to edit or click **+ Add** to create a new one.

> 2. Give your edge firewall a unique and easy-to-remember name.

Reviewer 2: Esses passos estão um pouco diferentes, é um novo padrão?

Author: então, com o novo guia de acesso ao RTM, a ideia é diminuirmos os passos. testei aqui! a @MarianaAguilera usou em alguns do marketplace pelo que lembro, também

Reviewer 2: É que como está fora da lista, parece que não é um passo.

Reviewer: deixei uma sugestão, tentando dar continuidade e manter o foco apenas no edit.

> [Access RTM](/en/documentation/products/guides/how-to-access-rtm/) > select **Edge Firewall**.
> 
> 1. From the list, select the edge firewall you want to edit or click **+ Add** to create a new one.
> 2. Give your edge firewall a unique and easy-to-remember name.

Reviewer: ```suggestion
[Access RTM](/en/documentation/products/guides/how-to-access-rtm/) > select **Edge Firewall**. Then, from the list, select the edge firewall you want to edit.
```

### Domains

> On the **Domains** section, you'll choose the domains associated to this edge firewall. This means that all configurations made in this edge firewall will be associated with the selected domains.

Reviewer: ```suggestion
In the **Domains** section, you'll choose the domains associated with this edge firewall. This means that all configurations made in this edge firewall will be associated with the selected domains.
```

1. On the **Available Domains** box, you can:
    - Type characters or words to find domains or scroll through the list of your domains.
    - Select each domain you want to associate to this data streaming and click the `>` button.
    - Select multiple domains by using **ctrl** on Windows and Linux or **command** on Mac and click the `>` button.
    - Select **Choose all** to select all domains presented on the list.
2. After you click the `>` button, the selected domains are moved to the **Chosen Domains** box.
    - You can also use the `<` button to remove specific domains already chosen or select **Clear selection** to remove all domains.

### Modules

[Modules](/en/documentation/products/edge-firewall/#about-edge-firewall-modules) can unlock additional functionalities and features related to your configured firewalls. See the [Pricing page](https://www.azion.com/en/pricing/) for more details.

There are four available modules that can be activated with your edge firewall:

> 

Reviewer: ```suggestion
```

> - <Badge variant="accent">[DDoS Protection Unmetered](/en/documentation/products/edge-firewall/ddos-protection/)</Badge>

Reviewer 2: Não usamos badge só para versionamento?

Author: era um teste também hehe pra não ficar tão sem graça..

Reviewer 2: Hahah eu sei, também quero mais customizações 😭

Reviewer: eu usei para produtos que precisam ser atualizados, achei similar o caso

Protects your content and applications against Distributed Denial of Service (DDoS) attacks. Available for the **Business** and higher Service Plans.

- <Badge variant="accent">[Edge Functions](/en/documentation/products/edge-firewall/edge-functions/firewall/)</Badge>

> Run serverless functions on Azion's Edge with low latency, bringing the operational power closer to the end user.

Reviewer: ```suggestion
Enables you to run serverless functions on Azion's edge with low latency, bringing the operational power closer to the end user.
```

Reviewer: o resto estava em 3ra pessoa.

- <Badge variant="accent">[Network Layer Protection](/en/documentation/products/edge-firewall/network-layer-protection/)</Badge>

> Allows the creation of filters by IP/CIDR, ASN addresses, or by countries (geolocation) through the configuration of Network Lists and the definition of business rules that will validate blocking or release Criteria.

Reviewer: ```suggestion
Allows the creation of filters by IP/CIDR, ASN addresses, or by countries (geolocation) through the configuration of Network Lists and the definition of business rules that will validate blocking or releasing Criteria.
```

- <Badge variant="accent">[Web Application Firewall](/en/documentation/products/edge-firewall/web-application-firewall/)</Badge>
Protects your applications against threats such as SQL Injections, Remote File Inclusion (RFI), Cross-Site Scripting (XSS), and many others.

You can choose between the list of modules: you can select one, a few, or even all.

### Status

> You'll also find a switch labeled **Active**. You can enable the switch to activate your edge firewall, and disable the switch if you no longer want this edge firewall acive.

Reviewer: ```suggestion
You'll also find a switch labeled **Active**. You can enable the switch to activate your edge firewall, and disable the switch if you no longer want this edge firewall active.
```

### Debug rules

You'll also find a switch labeled **Debug Rules**. If you enable the switch, you'll turn on the feature that allows you to check if rules you've created using Rules Engine for Edge Firewall have been successfully executed in your application.

<Button href="/en/documentation/products/guides/debug-rules/" text="go to How to debug rules guide" variant="secondary"></Button>

Once you're done configuring your main settings, click the **Save** button.

> 
> ## Via API

Reviewer: ```suggestion

---

## Via API
```

> 1. Run the following `GET` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) to retrieve your `<edge_application_id>`:

Reviewer 2: ```suggestion
1. Run the following `GET` request in your terminal, replacing `[TOKEN VALUE]` with your [personal token](/en/documentation/products/guides/personal-tokens/) to retrieve your `<edge_firewall_id>`:
```

```bash
curl --location 'https://api.azionapi.net/edge_firewall \
--header 'Accept: application/json; version=3' \
--header 'Authorization: Token [TOKEN VALUE]'
```

2. You'll receive a response with all your existing edge firewalls. Copy the value of the `<id>` that you want to configure.
3. Run a `PATCH` request to modify the firewall as follows:

```bash
curl --location --request PATCH 'https://api.azionapi.net/edge_firewall/<edge_firewall_id>' \
--header 'Accept: application/json; version=3' \
--header 'Authorization: Token [TOKEN VALUE]' \
--header 'Content-Type: application/json' \
--data '{
    "name": "My edge firewall",
    "domains": [
        <domains_id>
    ],
    "is_active": true,
        "edge_functions_enabled": true,
        "network_protection_enabled": true,
        "waf_enabled": true,
    "debug_rules": true
}'
```

  | Key | Description |
  | --- | --- |

>   | `name` | Name of this edge firewall |

Reviewer 2: ```suggestion
  | `name` | Name of the edge firewall |
```

>   | Key | Description |
>   | --- | --- |
>   | `name` | Name of the edge firewall |
>   | `domains` | Array of IDs of the domains you want to associated to this edge firewall |
>   | `is_active` | Boolean value that enables (`true`) or disables (`false`) the edge firewall |
>   | `edge_functions_enabled` | Boolean value that enables (`true`) or disables (`false`) the [Edge Functions](/en/documentation/products/edge-firewall/edge-functions/firewall/) module |
>   | `network_protection_enabled` | Boolean value that enables (`true`) or disables (`false`) the [Network Layer Protection](/en/documentation/products/edge-firewall/network-layer-protection/) module |
>   | `waf_enabled` | Boolean value that enables (`true`) or disables (`false`) the [Web Application Firewall (WAF)](/en/documentation/products/edge-firewall/web-application-firewall/) module |
>   | `debug_rules` | Boolean value that enables (`true`) or disables (`false`) the [Debug Rules](/en/documentation/products/guides/debug-rules/) feature |

Reviewer: ```suggestion
| Key | Description |
| --- | --- |
| `name` | Name of the edge firewall |
| `domains` | Array of IDs of the domains you want to associate with this edge firewall |
| `is_active` | Boolean value that enables (`true`) or disables (`false`) the edge firewall |
| `edge_functions_enabled` | Boolean value that enables (`true`) or disables (`false`) the [Edge Functions](/en/documentation/products/edge-firewall/edge-functions/firewall/) module |
| `network_protection_enabled` | Boolean value that enables (`true`) or disables (`false`) the [Network Layer Protection](/en/documentation/products/edge-firewall/network-layer-protection/) module |
| `waf_enabled` | Boolean value that enables (`true`) or disables (`false`) the [Web Application Firewall (WAF)](/en/documentation/products/edge-firewall/web-application-firewall/) module |
| `debug_rules` | Boolean value that enables (`true`) or disables (`false`) the [Debug Rules](/en/documentation/products/guides/debug-rules/) feature |
```

4. You'll receive a response similar to this:

```json
{
    "results": {
        "name": My edge firewall",
        "waf_enabled": true,
        "is_active": true,
        "edge_functions_enabled": true,
        "last_modified": "2023-06-01T18:00:11.994514Z",
        "domains": [
            0123456789
        ],
        "last_editor": "last.editor@azion.com",
        "network_protection_enabled": true,
        "id": 12345,
        "debug_rules": false
    },
    "schema_version": 3
}
```

Wait a few minutes for the changes to propagate and your edge firewall will be updated.

:::tip
Check the [Azion API documentation](https://api.azion.com/) and the [OpenAPI specification](https://github.com/aziontech/azionapi-openapi/) to know more about what the Azion API can offer.
:::

> 
> 

Reviewer: ```suggestion
```

---
title: How to configure main settings
description: Customize your edge firewall domains and modules.

> meta_tags: secure, edge firewal, configuration, settings, domains

Reviewer 2: ```suggestion
meta_tags: secure, edge firewall, configuration, settings, domains
```

namespace: docs_guides_secure_configure_main_settings
permalink: /documentation/products/guides/secure/configure-main-settings/
menu_namespace: secureMenu
---

import Button from '~/components/Button.astro'
import Badge from '~/components/Badge.astro'

> Once you create an edge firewall with Azion, you can modify a few configurations. In this guide, you'll learn to modify the domains associated to your firewall and to enable the available modules.

Author: ```suggestion
Once you create an edge firewall with Azion, you can modify a few configurations. In this guide, you'll learn to modify the domains associated with your firewall and to enable the available modules.
```

---

## Via RTM

In this section, you'll modify the main settings of your edge firewalls: name, domains, and modules via [Real-Time Manager (RTM)](https://manager.azion.com).

[Access RTM](/en/documentation/products/guides/how-to-access-rtm/) > select **Edge Firewall**.

1. From the list, select the edge firewall you want to edit or click **+ Add** to create a new one.
2. Give your edge firewall a unique and easy-to-remember name.

### Domains

In the **Domains** section, you'll choose the domains associated with this edge firewall. This means that all configurations made in this edge firewall will be associated with the selected domains.

1. On the **Available Domains** box, you can:
    - Type characters or words to find domains or scroll through the list of your domains.

>     - Select each domain you want to associate to this data streaming and click the `>` button.

Reviewer 3: ```suggestion
    - Select each domain you want to associate with this data streaming and click the `>` button.
```

- <Badge variant="accent">[Edge Functions](/en/documentation/products/edge-firewall/edge-functions/firewall/)</Badge>
Enables you to run serverless functions on Azion's edge with low latency, bringing the operational power closer to the end user.

- <Badge variant="accent">[Network Layer Protection](/en/documentation/products/edge-firewall/network-layer-protection/)</Badge>
Allows the creation of filters by IP/CIDR, ASN addresses, or by countries (geolocation) through the configuration of Network Lists and the definition of business rules that will validate blocking or release Criteria.

- <Badge variant="accent">[Web Application Firewall](/en/documentation/products/edge-firewall/web-application-firewall/)</Badge>
Protects your applications against threats such as SQL Injections, Remote File Inclusion (RFI), Cross-Site Scripting (XSS), and many others.

You can choose between the list of modules: you can select one, a few, or even all.

### Status

You'll also find a switch labeled **Active**. You can enable the switch to activate your edge firewall, and disable the switch if you no longer want this edge firewall active.

### Debug rules

> You'll also find a switch labeled **Debug Rules**. If you enable the switch, you'll turn on the feature that allows you to check if rules you've created using Rules Engine for Edge Firewall have been successfully executed in your application.

Reviewer 3: ```suggestion
You'll also find a switch labeled **Debug Rules**. If you enable the switch, you'll turn on the feature that allows you to check if the rules you've created using Rules Engine for Edge Firewall have been successfully executed in your application.
```
